tangelo/linq/target/backend.py

```diff
@@ -205,10 +207,18 @@ def simulate(self, source_circuit, return_statevector=False, initial_statevector
             numpy.array: The statevector, if available for the target backend
                 and requested by the user (if not, set to None).
         """
+        n_meas = source_circuit.counts.get("MEASURE", 0)
+
         if source_circuit.is_mixed_state and not self.n_shots:
             raise ValueError("Circuit contains MEASURE instruction, and is assumed to prepare a mixed state."
                              "Please set the n_shots attribute to an appropriate value.")
 
+        if desired_meas_result is not None:
+            if len(desired_meas_result) != n_meas or not isinstance(desired_meas_result, str):
+                raise ValueError("desired_meas result is not a string with the same length as the number of measurements"
```

Comment:
This should be 
```python3
if not isinstance(desired_meas_result, str) or len(desired_meas_result) != n_meas:
     raise ValueError(f"desired_meas_result of {desired_meas_result} is not a string with the same length as the number of measurements = {n_meas}"
```
as it will throw an error if someone inputs an object that does not have have a `len`.

You may also want a test of this in `test_simulator`

Reply:
ok